WPFHospitalEditorUnitTests/SearchAppointmentTest.cs

```diff
@@ -0,0 +1,289 @@
+        private Doctor doctor2;
+        private Doctor doctor3;
+        private TimeInterval timeInterval1;
+        private TimeInterval timeInterval2;
+        private TimeInterval timeInterval3;
+        private Room room1;
+        private Room room2;
+        private Shift shift1;
+        private Shift shift2;
+        private Shift shift5;
+        private RecommendationRequestDto dto1;
+        private RecommendationRequestDto dto2;
+        private RecommendationRequestDto dto3;
+        private RecommendationRequestDto dto4;
+        private RecommendationRequestDto dto5;
+        private RecommendationRequestDto dto6;
+        private List<Shift> shifts1 = new List<Shift>();
+        private List<Shift> shifts2 = new List<Shift>();
+        private List<Shift> shifts3 = new List<Shift>();
+
+        public SearchAppointmentTest()
+        {
+            
+            examinationService = new ExaminationService(mockExaminationRepository.Object, mockShiftRepository.Object, mockDoctorRepository.Object);
+
+
+            InitDoctors();
+            InitTimeInterval();
+            InitRooms();
+            InitDoctorShifts();
+            InitRecommendationDto();
+        }
+
+        [Fact]
+        public void Find_5_appointments_time_priority()
+        {
+
+            StubRepositories();
+            List<RecommendationDto> availableAppointments = new List<RecommendationDto>();
+
+            examinationService.Recommend(dto1).Count.ShouldBe(5);
+        }
+
+        [Fact]
+        public void Find_3_appointments_time_priority()
+        {
+            StubRepositories();
+            List<RecommendationDto> availableAppointments = new List<RecommendationDto>();
+            
+            examinationService.Recommend(dto2).Count.ShouldBe(3);
+        }
+
+        [Fact]
+        public void Find_0_appointments_time_priority()
+        {
+            StubRepositories();
+            List<RecommendationDto> availableAppointments = new List<RecommendationDto>();
+
+            examinationService.Recommend(dto3).Count.ShouldBe(0);
+        }
+
+        [Fact]
+        public void Find_5_appointments_with_doctor_priority()
+        {
+            StubRepositories();
+            List<RecommendationDto> availableAppointments = new List<RecommendationDto>();
+
+            examinationService.Recommend(dto4).Count.ShouldBe(5);
+        }
+
+        [Fact]
+        public void Find_3_appointments_with_doctor_priority()
+        {
+            StubRepositories();
+            List<RecommendationDto> availableAppointments = new List<RecommendationDto>();
+
+            examinationService.Recommend(dto5).Count.ShouldBe(3);
+        }
+
+        [Fact]
+        public void Find_0_appointments_with_doctor_priority()
+        {
+            StubRepositories();
+            List<RecommendationDto> availableAppointments = new List<RecommendationDto>();
+
+            examinationService.Recommend(dto6).Count.ShouldBe(0);
+        }
```

Comment:
All test are identical. They are different only in the dto data that is sent and in the expected response. Use [THEORY ](https://www.youtube.com/watch?v=SqI1BfYg0No&feature=youtu.be&ab_channel=NikolaLuburi%C4%87)to avoid unnecessary code copying.

Reply:
Fixed